visualization/app/codeCharta/ui/ribbonBar/ribbonBar.component.ts

```diff
@@ -2,26 +2,17 @@ import "./ribbonBar.component.scss"
 import $ from "jquery"
 
 export class RibbonBarController {
-	private collapsingElements = $("ribbon-bar-component .element-to-toggle")
-	private isExpanded: boolean = false
+	private readonly EXPANDED_CLASS = "expanded"
 
-	public toggle() {
-		if (this.isExpanded) {
-			this.collapse()
+	public toggle(event) {
+		const boxElement = $(event.srcElement).closest("md-card")
+
+		if (boxElement.hasClass(this.EXPANDED_CLASS)) {
+			boxElement.removeClass(this.EXPANDED_CLASS)
 		} else {
-			this.expand()
+			boxElement.addClass(this.EXPANDED_CLASS)
 		}
 	}
```

Comment:
@alexhunziker I thought about using ng-class, but this approach is more generic, as it can handle every of the 4 sections according to its relative html-nodes without declaring them or passing string parameters through the `toggle()` function.